### PR TITLE
[Vue] migrate popover handler to popover service

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR depends on #18443.**

Changes:
* The current popover handler directive which if placed on a page handles the popover parameter has been replaced with a service class that does the same thing but doesn't need any HTML placed on the page.

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
UI tests should be passing for this one

**Comment by peterhashair** [comment]
maybe we need to resolve the conflicts, I am trying to fix those local, but it seems to return a blank page without any javascript errors on my tests.

File: plugins/CoreHome/vue/src/AjaxForm/AjaxForm.vue [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    return {
+      isSubmitting: false,
+      successfulPostResponse: null,
+      errorPostResponse: null,
+    };
+  },
+  emits: ['update:modelValue'],
+  mounted() {
+    // on submit call controller submit method
+    $(this.$refs.root).on('click', 'input[type=submit]', () => {
```
Comment (peterhashair) [inline plugins/CoreHome/vue/src/AjaxForm/AjaxForm.vue]: just wondering, will $emit function works here.

**Review by peterhashair — COMMENTED** [review]
just one suggestion, if that works

File: plugins/CoreHome/vue/src/AjaxForm/AjaxForm.vue (hunk 1 above)
Comment (diosmosis) [reply plugins/CoreHome/vue/src/AjaxForm/AjaxForm.vue]: I'm not sure what you mean, how would $emit be used here?

Comment (peterhashair) [reply plugins/CoreHome/vue/src/AjaxForm/AjaxForm.vue:69]: ah, actually i confused myself, I was thinking this 
```html
 <div ref="root" @click=‘this.submitForm()’>
```

Comment (peterhashair) [reply plugins/CoreHome/vue/src/AjaxForm/AjaxForm.vue]: ah, actually I confused myself, I see why this way.

**Pull request merged**